tools/emr/submit_datagen_job.py

```diff
@@ -41,11 +42,11 @@
     ec2_instances = [dict(row) for row in reader]
 
 
-def calculate_cluster_config(scale_factor, sf_ratio=100.0):
+def calculate_cluster_config(scale_factor, sf_ratio):
     num_workers = max(min_num_workers, min(max_num_workers, ceil(scale_factor / sf_ratio)))
     return {
         'num_workers': num_workers,
-        'parallelism_factor': max(1.0, sf_ratio / 100.0)
+        'parallelism_factor': max(1.0, sf_ratio / 50.0)
```

Comment:
Using sf_ratio to control the `num_workers` is incorrect in the form, as it was meant to tune the worker count and per-worker parallelism ratio.
low sf_ratio -> lot of workers with small parallelism
high sf_ratio -> few workers with large parallelism
in the case of choosing a larger/smaller instance, we want to tune the number of workers irrespective of the parallelism_factor as the larger vcpu count will deal with the parallelism anyways. a simple fix is to add a vcpu parameter to this function, and calculate this line as
```suggestion
        'parallelism_factor': max(1.0, sf_ratio / vcpu / 6.0) 
```
to refactor more, i think we could ditch line 104 altogether and calculate `num_threads` here directly, it would be cleaner